src/main/java/com/ss/mqtt/broker/model/topic/AbstractTopic.java

```diff
@@ -15,41 +16,32 @@
         DebugUtils.registerIncludedFields("rawTopic");
     }
 
-    static final String DELIMITER = "/";
-    static final String MULTI_LEVEL_WILDCARD = "#";
-    static final String SINGLE_LEVEL_WILDCARD = "+";
-
-    static void checkTopic(@NotNull String topic) {
-        if (topic.length() == 0) {
-            throw new IllegalArgumentException("Topic has zero length.");
-        } else if (topic.contains("//") || topic.startsWith("/") || topic.endsWith("/")) {
-            throw new IllegalArgumentException("Topic has zero length level: " + topic);
-        }
-    }
-
-    private final @NotNull String[] segments;
+    private final @NotNull Array<String> segments;
```

Comment:
I think any topic is immutable so better to use native array instead of any collection here

Reply:
Fixed